WooCommerce/WooCommerceTests/ViewRelated/Blaze/BlazeCampaignCreationFormViewModelTests.swift

```diff
@@ -0,0 +1,493 @@
+                                       BlazeAISuggestion(siteName: "Second suggested tagline", textSnippet: "Second suggested description"),
+                                       BlazeAISuggestion(siteName: "Third suggested tagline", textSnippet: "Third suggested description")]
+
+    /// Mock Storage: InMemory
+    private var storageManager: StorageManagerType!
+
+    /// View storage for tests
+    private var storage: StorageType {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        super.setUp()
+        storageManager = MockStorageManager()
+    }
+
+    // MARK: Initial values
+    func test_image_is_empty_initially() async throws {
+        // Given
+        let viewModel = BlazeCampaignCreationFormViewModel(siteID: sampleSiteID,
+                                                           productID: sampleProductID,
+                                                           onCompletion: {})
+
+        // Then
+        XCTAssertNil(viewModel.image)
+    }
+
+    func test_tagline_is_empty_initially() async throws {
+        // Given
+        let viewModel = BlazeCampaignCreationFormViewModel(siteID: sampleSiteID,
+                                                           productID: sampleProductID,
+                                                           onCompletion: {})
+
+        // Then
+        XCTAssertEqual(viewModel.tagline, "")
+    }
+
+    func test_description_is_empty_initially() async throws {
+        // Given
+        let viewModel = BlazeCampaignCreationFormViewModel(siteID: sampleSiteID,
+                                                           productID: sampleProductID,
+                                                           onCompletion: {})
+
+        // Then
+        XCTAssertEqual(viewModel.description, "")
+    }
+
+    // MARK: Download product image
+
+    func test_it_reads_product_from_storage_for_displaying_image() async throws {
+        // Given
+        let sampleProductImage = ProductImage.fake().copy(imageID: 1)
+        insertProduct(.fake().copy(siteID: sampleSiteID,
+                                   productID: sampleProductID,
+                                   statusKey: (ProductStatus.published.rawValue),
+                                   images: [sampleProductImage]))
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        stores.whenReceivingAction(ofType: BlazeAction.self) { [weak self] action in
+            guard let self = self else { return }
+            switch action {
+            case let .fetchAISuggestions(_, _, completion):
+                completion(.success(sampleAISuggestions))
+            default:
+                XCTFail("Unexpected action: \(action)")
+            }
+        }
+        let imageLoader = MockProductUIImageLoader()
+        let sampleImage = UIImage.addOutlineImage
+        imageLoader.requestImageStubbedResponse = sampleImage
```

Comment:
Suggestion: This setup can be moved to reusable helper methods to keep the tests shorter and more readable. E.g: `mockAISuggestionRequest`, `mockImageDownload`, `mockProductInStorage`.